src/engine/utils/DirPath.cpp

```diff
@@ -65,3 +70,60 @@ ButtonPath::ButtonPath(const std::string& filename)
 {
     (*this) << "Resources" << "Buttons" << filename;
 }
+
+namespace {
+#if defined(LINUX)
+
+  // look up my current home directory in /etc/passwd (or NIS or
+  // wherever).
+  std::unique_ptr<std::string> find_pwd_home()
+  {
+    size_t bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+    if(bufsize == static_cast<size_t>(-1))
+      bufsize = 16384;		// Should be big enough
+
+    struct passwd pwd;
+    std::vector<char> buf(bufsize);
+    struct passwd* result;
+
+    auto err = getpwuid_r(getuid(), &pwd, buf.data(), buf.size(), &result);
+    if(!result)
+      {
+	if(err == 0)
+	  g_LogFile.warning("io", "User has no home directory.");
+	else
+	  g_LogFile.error("io", "getpwuid_r() failed: ", std::strerror(err), " (", err, ")");
+
+	return {};
+      }
+
+    return std::make_unique<std::string>(pwd.pw_dir);
+  }
+
+  // Expand "~/some/dir" to "<my home-dir>/some/dir"
+  //
+  // TODO: Expand "~mike/dir" to "<mike's home-dir>/dir"
+  std::string tildeexpand(std::string path)
+  {
+    if(path.empty())
+      return path;
+
+    if(path.substr(0, 2) == "~/")
```

Comment:
could you please use braces if the body of the if spans multiple lines? I think that makes it less error prone

Reply:
Fixed. And thanks for pointing it out.

Reply:
Huh, you fixed it too?

Oh, well... :+1: